core/src/HIP/Kokkos_HIP_IsXnack.hpp

```diff
@@ -0,0 +1,52 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+/*--------------------------------------------------------------------------*/
+
+#ifndef KOKKOS_HIP_ISXNACK_HPP
+#define KOKKOS_HIP_ISXNACK_HPP
+#ifndef KOKKOS_IMPL_PUBLIC_INCLUDE
+#define KOKKOS_IMPL_PUBLIC_INCLUDE
+#endif  // KOKKOS_IMPL_PUBLIC_INCLUDE
+
+namespace Kokkos::Impl {
+
+/* Returns true iff we think the AMD GPU can access allocations created with the
+system allocator.
```

Comment:
For now we want system allocations to be available only for the APU. I Maybe we can spell that out in the comment?